db/migrate/20160419134542_add_deployment_authentication_options_to_authentications.rb

```diff
@@ -0,0 +1,35 @@
+class AddDeploymentAuthenticationOptionsToAuthentications < ActiveRecord::Migration[5.0]
+  def change
+    add_column :authentications, :auth_challenge, :boolean
+    add_column :authentications, :auth_login, :boolean
+    # from old table t.string :kind assuming we could use auth_type is the correct? (could be ldap/htpass etc..)
+    add_column :authentications, :auth_public_key, :text #for supporting ssh public key (needed on deployment model)
+    add_column :authentications, :htpassd_users, :text, :array => true, :default => []
+    add_column :authentications, :ldap_id, :text, :array => true, :default => []
+    add_column :authentications, :ldap_email, :text, :array => true, :default => []
+    add_column :authentications, :ldap_preferred_user_name, :text, :array => true, :default => []
+    add_column :authentications, :ldap_bind_dn, :string
+    add_column :authentications, :ldap_bind_password, :string #should i store this as password?
+    add_column :authentications, :ldap_ca, :text #should i store this as auth_key?
+    add_column :authentications, :ldap_insecure, :boolean
+    add_column :authentications, :ldap_url, :string
+    add_column :authentications, :request_header_challenge_url, :string
+    add_column :authentications, :request_header_login_url, :string
+    add_column :authentications, :request_header_client_ca, :text #should i store this as auth_key?
+    add_column :authentications, :request_header_headers, :text, :array => true, :default => []
+    add_column :authentications, :request_header_preferred_username_headers, :text, :array => true, :default => []
+    add_column :authentications, :request_header_name_headers, :text, :array => true, :default => []
+    add_column :authentications, :request_header_email_headers, :text, :array => true, :default => []
+    add_column :authentications, :open_id_client_id, :string #should i store it in user_id?
+    add_column :authentications, :open_id_client_secret, :string #should i store this as password?
+    add_column :authentications, :open_id_sub_claim, :string
+    add_column :authentications, :open_id_user_info, :string
+    add_column :authentications, :open_id_ca, :text #should i store this as auth_key?
+    add_column :authentications, :open_id_authorization_endpoint, :string
+    add_column :authentications, :open_id_token_endpoint, :string
+    add_column :authentications, :open_id_extra_scopes, :text, :array => true, :default => []
+    add_column :authentications, :open_id_extra_authorize_parameters, :text
+    add_column :authentications, :google_hosted_domain, :string
+    add_column :authentications, :github_organizations, :text, :array => true, :default => []
+  end
```

Comment:
Oh I didn't realize they were public...in that case they can go to auth_key_public, or perhaps a new column that's just `certificate` or `certificate_authority` or `ca` (If you haven't noticed I prefer not using abbreviations 😜 )